### PR TITLE
Feature/keyboard shortcuts

### PR DESCRIPTION
**Pull request by sternomer**

### CODE REVIEW

File: src/store/modules/loading.js [hunk 1]
```diff
@@ -1,13 +1,15 @@
 const state = {
   onGoingRequests: 0,
   isLoading: false,
+  popupStatus: false,
```
Comment (sternomer) [inline src/store/modules/loading.js:4]: better name: isPopupOpen

File: src/components/popups/share-popup/SharePopup.vue [hunk 2]
```diff
@@ -51,21 +51,27 @@ export default {
...
+        this.isShareMode = val;
+        this.$store.commit('changePopupStatus')
+      } 
     },
   },
   props: ["files"],
   methods: {
     async open() {
-      this.dialog = true;
+      if (!this.$store.getters.popupStatus) {
```
Comment (sternomer) [inline src/components/popups/share-popup/SharePopup.vue:63]:
```suggestion
      if (this.$store.getters.popupStatus) {
          return;
      }
```

Comment (sternomer) [inline src/components/popups/share-popup/SharePopup.vue:63]: fix formatting

```suggestion
      if (this.$store.getters.popupStatus) {
          return
      }
```

File: src/components/popups/share-popup/SharePopup.vue [hunk 3]
```diff
@@ -51,21 +51,27 @@ export default {
...
+        this.$store.commit('changePopupStatus')
```
Comment (sternomer) [inline src/components/popups/share-popup/SharePopup.vue:64]: move this to watch

File: src/components/popups/share-popup/SharePopup.vue [hunk 4]
```diff
@@ -51,21 +51,27 @@ export default {
   components: { Share, EditPermissions, CopyPopup },
   watch: {
     dialog: function (val) {
-      if (!val) this.isShareMode = val;
+      if (!val) {
+        this.isShareMode = val;
+        this.$store.commit('changePopupStatus')
```
Comment (sternomer) [inline src/components/popups/share-popup/SharePopup.vue:56]: move this out of the if (so it works for both open and close)

File: src/components/popups/info-popup/BaseInfoPopup.vue [hunk 5]
```diff
@@ -63,18 +63,26 @@ export default {
       externalUsersFailed: [],
     };
   },
+  watch: {
+    dialog(val) {
+      if (!val) {
+        this.$store.commit('changePopupStatus')
```
Comment (sternomer) [inline src/components/popups/info-popup/BaseInfoPopup.vue:69]: same

File: src/components/popups/info-popup/BaseInfoPopup.vue [hunk 6]
```diff
@@ -63,18 +63,26 @@ export default {
...
+      } 
+    },
+  },
   methods: {
     async open() {
-      let externalPermissionsRes = [];
-      this.externalUsers = [];
-      this.externalUsersFailed = [];
+      if (!this.$store.getters.popupStatus) {
```
Comment (sternomer) [inline src/components/popups/info-popup/BaseInfoPopup.vue:75]: same

File: src/components/popups/info-popup/BaseInfoPopup.vue [hunk 7]
```diff
@@ -86,9 +94,11 @@ export default {
 
         this.externalUsersFailed = externalUsersFailed;
         this.externalUsers = externalUsers;
-      }
+        }
 
-      this.dialog = true;
+        this.dialog = true;
+        this.$store.commit('changePopupStatus')
```
Comment (sternomer) [inline src/components/popups/info-popup/BaseInfoPopup.vue:100]: same

File: src/components/popups/external-transfer-popup/ExternalTransferPopup.vue [hunk 8]
```diff
@@ -144,8 +147,11 @@ export default {
   },
   methods: {
     open() {
-      this.dialog = true;
-      this.currentStep = 1;
+      if (!this.$store.getters.popupStatus) {
```
Comment (sternomer) [inline src/components/popups/external-transfer-popup/ExternalTransferPopup.vue:150]: same

File: src/components/popups/external-transfer-popup/ExternalTransferPopup.vue [hunk 9]
```diff
@@ -134,7 +134,10 @@ export default {
   },
   props: ["file"],
   watch: {
-    dialog() {
+    dialog(val) {
+      if (!val) {
```
Comment (sternomer) [inline src/components/popups/external-transfer-popup/ExternalTransferPopup.vue:138]: same

File: src/components/popups/Preview.vue [hunk 10]
```diff
@@ -63,10 +63,20 @@ export default {
...
+      if (!val) {
+        this.$store.commit('changePopupStatus')
+      } 
+    },
+  },
   methods: {
     open(file) {
-      this.file = file;
-      this.dialog = true;
+      if (!this.$store.getters.popupStatus) {
```
Comment (sternomer) [inline src/components/popups/Preview.vue:75]: same

File: src/components/popups/Preview.vue [hunk 11]
```diff
@@ -63,10 +63,20 @@ export default {
       file: undefined,
     };
   },
+  watch: {
+    dialog(val) {
+      if (!val) {
```
Comment (sternomer) [inline src/components/popups/Preview.vue:68]: same

File: src/components/popups/MoveToPopup.vue [hunk 12]
```diff
@@ -46,11 +46,21 @@ export default {
...
+        this.$store.commit('changePopupStatus')
+      } 
+    },
+  },
   props: ["files"],
   methods: {
     async open() {
-      await Promise.all([this.fetchHierachy(this.currentFolder), this.fetchFolders(this.currentFolder)]);
-      this.dialog = true;
+      if (!this.$store.getters.popupStatus) {
```
Comment (sternomer) [inline src/components/popups/MoveToPopup.vue:59]: same

File: src/components/popups/MoveToPopup.vue [hunk 13]
```diff
@@ -46,11 +46,21 @@ export default {
       folderDest: undefined,
     };
   },
+  watch: {
+    dialog(val) {
+      if (!val) {
+        this.$store.commit('changePopupStatus')
```
Comment (sternomer) [inline src/components/popups/MoveToPopup.vue:52]: same

File: src/components/popups/BaseNamePopup.vue [hunk 14]
```diff
@@ -55,7 +62,10 @@ export default {
       return this.name.length < 2 || !isValidString(this.name);
     },
     open() {
-      this.dialog = true;
+      if (!this.$store.getters.popupStatus) {
```
Comment (sternomer) [inline src/components/popups/BaseNamePopup.vue:65]: same

File: src/components/popups/BaseNamePopup.vue [hunk 15]
```diff
@@ -46,6 +46,13 @@ export default {
       name: "",
     };
   },
+  watch: {
+    dialog(val) {
+      if (!val) {
```
Comment (sternomer) [inline src/components/popups/BaseNamePopup.vue:51]: same

File: src/components/popups/BaseAlertPopup.vue [hunk 16]
```diff
@@ -24,11 +24,21 @@ export default {
...
+        this.$store.commit('changePopupStatus')
+      } 
+    },
+  },
   props: ["img", "text", "button", "cancelButton"],
   methods: {
     open(data) {
-      this.data = data;
-      this.dialog = true;
+      if (!this.$store.getters.popupStatus) {
```
Comment (sternomer) [inline src/components/popups/BaseAlertPopup.vue:37]: same

File: src/components/popups/BaseAlertPopup.vue [hunk 17]
```diff
@@ -24,11 +24,21 @@ export default {
       data: undefined,
     };
   },
+  watch: {
+    dialog(val) {
+      if (!val) {
+        this.$store.commit('changePopupStatus')
```
Comment (sternomer) [inline src/components/popups/BaseAlertPopup.vue:30]: same

**Review by sternomer — COMMENTED** [review]
Check again if there are more popups that need to watch for "dialog" changes (not open when others are open) example: CreateFilePopup

**Review by MichaelSimkin — CHANGES_REQUESTED** [review]
Looks good. 
One last thing: I think the store functions for the popup logic don't belong in the "loading" store. You should create a new store for them.

**Review by MichaelSimkin — APPROVED** [review]

**Pull request merged**